tests/test_agpypeline.py

```diff
@@ -0,0 +1,487 @@
+#!/usr/bin/env python3
+"""
+Purpose: Unit testing for agpypeline files
+Author : Chris Schnaufer <schnaufer@arizona.edu>
+Notes:
+    This file assumes it's in a subfolder off the main folder
+"""
+import argparse
+import json
+import os
+import piexif
+
+import numpy as np
+
+from numpy import nan
+from liblas.header import Header
+from osgeo import gdal, ogr, osr
+from shapely.geometry import Point, LineString, MultiPoint
+
+from agpypeline import algorithm
+from agpypeline import configuration
+from agpypeline import entrypoint
+from agpypeline import environment
+from agpypeline import geoimage
+from agpypeline import geometries
+from agpypeline import lasfile
+
+
+TEST_FILES = ['agpypeline/algorithm.py', 'agpypeline/configuration.py',
+              'agpypeline/entrypoint.py', 'agpypeline/environment.py', 'agpypeline/geoimage.py',
+              'agpypeline/geometries.py', 'agpypeline/lasfile.py']
+
+JSON_FILE = os.path.realpath('data/geometries.json')
+
+TEST_IMAGE = os.path.realpath('images/output.tin.tif')
+
+TEST_METADATA = os.path.realpath('data/08f445ef-b8f9-421a-acf1-8b8c206c1bb8_metadata_cleaned.json')
+
+
+def test_exists():
+    """Tests if the files stored in variables above are accessible/contained"""
+    for file in TEST_FILES:
+        assert os.path.isfile(file)
+    assert os.path.isfile(TEST_IMAGE) and os.path.isfile(TEST_METADATA)
+
+
+# Other methods can be used here if wanted in order to test if the error is raised, but this would
+# require creating a class
+def test_algorithm():
+    """Tests initializing the algorithm.Algorithm() class within agpypeline and performing a process with default
+    parameters """
+    alg = algorithm.Algorithm()
+    try:
+        alg.perform_process(environment.Environment(configuration.Configuration()), {}, {}, [])
+        assert True is False
+    except RuntimeError as error:
+        assert str(error) == "The Algorithm class method perform_process() must be overridden by a derived class"
+
+
+def test_configuration():
+    """Tests agpypeline's configuration.Configuration() class by initializing configuration.Configuration() and
+     checking to make sure that all necessary parameters are contained"""
+    config = configuration.Configuration
+    for entry in ['transformer_version', 'transformer_description', 'transformer_name', 'transformer_sensor',
+                  'transformer_type', 'author_name', 'author_email', 'contributors', 'repository']:
+        assert hasattr(config, entry)
+        if entry == 'contributors':
+            assert getattr(config, entry) == []
+        else:
+            assert getattr(config, entry) is None
+
+
+def test_entrypoint_handle_error():
+    """Tests entrypoint's handle_error function by passing in an error code and message"""
+    entry = entrypoint.__internal__()
+    bad_result = entry.handle_error(None, None)
```

Comment:
The data used here could be loaded from a JSON file - including the expected results. if run in a loop it's easy to extend the testing as needed